precompile/fee_config_manager.go

```diff
@@ -0,0 +1,268 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package precompile
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/commontype"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	// must preserve order of these fields
+	gasLimitKey = iota
+	targetBlockRateKey
+	minBaseFeeKey
+	targetGasKey
+	baseFeeChangeDenominatorKey
+	minBlockGasCostKey
+	maxBlockGasCostKey
+	blockGasCostStepKey
+
+	minFeeConfigFieldKey = gasLimitKey
+	maxFeeConfigFieldKey = blockGasCostStepKey
+
+	numFeeConfigField = (maxFeeConfigFieldKey - minFeeConfigFieldKey + 1)
+)
+
+var (
+	_ StatefulPrecompileConfig = &FeeConfigManagerConfig{}
+
+	// Singleton StatefulPrecompiledContract for setting fee configs by permissioned callers.
+	FeeConfigManagerPrecompile StatefulPrecompiledContract = createFeeConfigManagerPrecompile(FeeConfigManagerAddress)
+
+	setFeeConfigSignature = CalculateFunctionSelector("setFeeConfig(uint256,uint256,uint256,uint256,uint256,uint256,uint256,uint256)")
+	getFeeConfigSignature = CalculateFunctionSelector("getFeeConfig()")
+
+	ErrCannotChangeFee = errors.New("non-enabled cannot change fee config")
+
+	// 8 fields in FeeConfig struct
+	feeConfigInputLen = common.HashLength * numFeeConfigField
+)
+
+// FeeConfigManagerConfig wraps [AllowListConfig] and uses it to implement the StatefulPrecompileConfig
+// interface while adding in the contract deployer specific precompile address.
+type FeeConfigManagerConfig struct {
+	AllowListConfig
+}
+
+// Address returns the address of the fee config manager contract.
+func (c *FeeConfigManagerConfig) Address() common.Address {
+	return FeeConfigManagerAddress
+}
+
+// Configure configures [state] with the desired admins based on [c].
+func (c *FeeConfigManagerConfig) Configure(state StateDB) {
+	c.AllowListConfig.Configure(state, FeeConfigManagerAddress)
+}
+
+// Contract returns the singleton stateful precompiled contract to be used for the fee manager.
+func (c *FeeConfigManagerConfig) Contract() StatefulPrecompiledContract {
+	return FeeConfigManagerPrecompile
+}
+
+// GetFeeConfigManagerStatus returns the role of [address] for the fee config manager list.
+func GetFeeConfigManagerStatus(stateDB StateDB, address common.Address) AllowListRole {
+	return getAllowListStatus(stateDB, FeeConfigManagerAddress, address)
+}
+
+// SetFeeConfigManagerStatus sets the permissions of [address] to [role] for the
+// fee config manager list. assumes [role] has already been verified as valid.
+func SetFeeConfigManagerStatus(stateDB StateDB, address common.Address, role AllowListRole) {
+	setAllowListRole(stateDB, FeeConfigManagerAddress, address, role)
+}
+
+// PackGetFeeConfigInput packs the getFeeConfig signature
+func PackGetFeeConfigInput() []byte {
+	return getFeeConfigSignature
+}
+
```

Comment:
Do we need a function for this?

Reply:
not actually, but prefer to follow our convention here :D